src/Microsoft.AspNet.Http.Abstractions/Authentication/AuthenticationManager.cs

```diff
@@ -11,9 +12,24 @@ public abstract class AuthenticationManager
     {
         public abstract IEnumerable<AuthenticationDescription> GetAuthenticationSchemes();
 
-        public abstract AuthenticationResult Authenticate(string authenticationScheme);
+        public abstract void Authenticate(AuthenticateContext context);
 
-        public abstract Task<AuthenticationResult> AuthenticateAsync(string authenticationScheme);
+        public abstract Task AuthenticateAsync(AuthenticateContext context);
```

Comment:
I'm not sure it's that uncommon: I've been asked at least 3 or 4 times on JabbR how to retrieve the expiration date of a cookie or a token.